src/io/limits.rs

```diff
@@ -0,0 +1,15 @@
+/// Limits on how much memory that can be used in the decoding process. These limits are "best
+/// effort" limits and there are no strict guarantees that they will be followed at all times.
+#[derive(Debug, Copy, Clone)]
+pub struct DecodingLimits {
```

Comment:
A limit on `width` and `height` would also make sense. That would also be supported by all formats.

Reply:
Prior art from libpng includes width/height in its limits, but it does not provide a rationale. It seems to do so for legacy reasons: https://libpng.sourceforge.io/decompression_bombs.html

Aside of width/height there is also frame count to consider for GIF and APNG, but I'm not convinced these need to be separate knobs and not a global limit on the in-memory size of the decompressed image.